src/appmixer/kafka/routes.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const connections = require('./connections');
+
+module.exports = (context) => {
+
+    context.http.router.register({
+        method: 'POST',
+        path: '/connect/{mode}',
+        options: {
+            handler: async (req, h) => {
+
+                const { flowId, componentId } = req.payload;
+                const { mode } = req.params;
+                if (mode === 'consumer') {
+                    await connections.removeConnection({ flowId, componentId }, mode);
+                }
+                await connections.addConnection(context, req.payload, mode);
+                if (mode === 'producer') {
+                    await connections.sendMessage({ flowId, componentId, payload: req.payload });
+                }
+                await context.service.stateSet(`${flowId}:${componentId}`, { ...req.payload, mode });
```

Comment:
This is redundant. You use this API to send every message. So with every message you set the `state` (to the same value). Plus you save the `message` to the state, which is not correct. If you send 10k messages, you will trigger 10k mongo updates, that do nothing.